crates/biome_service/src/explain/rules.rs

```diff
@@ -0,0 +1,1692 @@
+//! Generated file, do not edit by hand, see `xtask/codegen`
+
+use biome_analyze::{FixKind, RuleMetadata};
+pub(super) fn get_rule_metadata(s: &str) -> Option<RuleMetadata> {
+    match s {
+        "noAccessKey" => Some(RuleMetadata {
+            name: "noAccessKey",
+            version: "1.0.0",
+            fix_kind: Some(FixKind::Unsafe),
+            recommended: true,
+            deprecated: None,
+            docs: NO_ACCESS_KEY,
+        }),
```

Comment:
The metadata is information that we already possess and it's stored statically by the compiler. We just need to extract it.

Ideally, something like that:

```rs
 #[derive(Default)]
    struct LintRulesVisitor {
        rules_metadata: BTreeMap<&'static str, &'static str>,
    }
    impl RegistryVisitor<JsLanguage> for LintRulesVisitor {
        fn record_category<C: GroupCategory<Language = JsLanguage>>(&mut self) {
            if matches!(C::CATEGORY, RuleCategory::Lint) {
                C::record_groups(self);
            }
        }

        fn record_rule<R>(&mut self)
        where
            R: Rule + 'static,
            R::Query: Queryable<Language = JsLanguage>,
            <R::Query as Queryable>::Output: Clone,
        {
            self.rules_metadata.insert(R::METADATA.name, R::METADATA.docs);
        }
    }
```

You can find similar code [here](https://github.com/biomejs/biome/blob/20d124626f1497b6d7b08d8140f19251118167fe/xtask/lintdoc/src/main.rs#L80-L105) and [here](https://github.com/biomejs/biome/blob/20d124626f1497b6d7b08d8140f19251118167fe/xtask/codegen/src/generate_configuration.rs#L20-L43).

Then you can add this function

```rs
    impl LintRulesVisitor {
        fn get_docs(&self, &'str) -> Option<&'static str> {
		self.rules_metadata.get(str)
	}
    }

```

So I believe there's nothing to generate, right?

Reply:
Before I address other comments I think it best to resolve this one first because it will probably have impact on those. Are we only interested in name and docs part of metadata?

The second approach you suggested here is essentially what I did but only with doc part of it or should `NO_DEBUGGER` be an already formated string with the other parts in there as well?

Witch approach would you prefer?

Reply:
Sorry for the delay in the response, I thought I had written the reply but I didn't hit the submit button 😞 

I prefer the first approach, because:
- it doesn't involve code generation
- it uses the information we already have

I don't think we will experience performance issues when retrieving static information.